src/components/notes/offline-indicator.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { Tooltip, Grid } from '@material-ui/core'
+import { Icon, Chip, useI18n } from 'cozy-ui/react'
+import useBreakpoints from 'cozy-ui/react/hooks/useBreakpoints'
+
+import styles from 'components/notes/offline-indicator.styl'
+
+/**
+ * Large tooltip, for browser view
+ *
+ * @param {object} props
+ * @param {boolean} props.open - true if we should force the tooltip to be opened
+ * @param {*} props.content - what to display inside the banner
+ * @param {*} props.children - the saving indicator
+ */
+function Balloon(props) {
+  // We add a dynamic `key`property to throw the old
+  // instance and get a new one if the `open` property changes.
+  // This is because we want to force the tooltip to be open
+  // for a few seconds (and MUI initializes the component to
+  // a controlled one) but we want the automatic (no-controlled)
+  // mode afterwards
+  return (
+    <Tooltip
+      title={props.content}
+      open={props.open || undefined}
+      key={props.open ? 'open' : 'auto'}
+      classes={{ tooltip: styles.tooltip }}
+    >
+      {props.children}
+    </Tooltip>
+  )
+}
+
+/**
+ * Banner on top of the editor, for mobile view
+ *
+ * @param {object} props
+ * @param {Ref} props.bannerRef - where to display the banner
+ * @param {*} props.content - what to display inside the banner
+ * @param {*} props.children - the saving indicator
+ */
+function Banner(props) {
+  const ref = props.bannerRef && props.bannerRef.current
+  if (ref) {
+    return ReactDOM.createPortal(
+      <div className={styles.banner}>{props.content}</div>,
+      ref
+    )
+  }
+  return null
+}
+
+/**
+ * Adds an indicator when offline
+ *
+ * It's mapped to a tooltip for browser views
+ * and to a banner for mobile views
+ * @param {object} props
+ * @param {boolean} props.open - true if we should force the tooltip to be opened
+ * @param {Ref} props.bannerRef - where to display the banner
+ * @param {*} props.children - the saving indicator
+ */
+export default function OfflineIndicator(props) {
+  const { t } = useI18n()
+  const { isMobile } = useBreakpoints()
+  const content = (
+    <Grid
+      direction="row"
+      container
+      wrap="nowrap"
+      spacing={16}
+      className={styles.grid}
+    >
```

Comment:
I'm wondering if this Grid should not be moved to UI. You're setting a 1rem padding and just for that, it should be centralized to UI.

Reply:
I am not convinced. This is the first use, and I really have no clue if other uses of a grid may need a 1rem padding. I would wait for a second or third use before trying to mutualize a technical component